### PR TITLE
Added Ui for Stay

### LINKED ISSUE #207: Show list of stay
I think this will be shown in room tab. By having a color indicator of vacant room.

### PR DESCRIPTION
**Pull request by sancharib-git**
image

### CODE REVIEW

File: src/main/java/seedu/address/model/Hotel.java [hunk 1]
```diff
@@ -438,4 +440,19 @@ public void addStay(Stay stay) {
...
+    /**
+     * Goes through stay list and sets
+     * the current stay for each room
+     */
+    public void updateRoomStays() {
+        for (Stay stay : stayList) {
+            for (Room room : roomList) {
+                if (stay.getRoom().equals(room)) {
+                    Room newRoom = new Room(room);
+                    newRoom.setStay(stay);
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java:454]: Can we just modify the stay in the old object?

File: src/main/java/seedu/address/model/hotel/room/Room.java [hunk 2]
```diff
@@ -34,6 +37,16 @@ public Room(String roomNum) {
...
+    /**
+     * Construct a room with attributes of @param other
+     */
+    public Room(Room room) {
+        this.roomTier = room.getTier();
+        this.roomNum = room.getRoomNum();
+        this.roomId = room.getRoomId();
+        this.roomCost = room.getRoomCost();
+    }
+
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/hotel/room/Room.java:49]: If we can modify the stay directly, we don't need this right?

**Review by dawidjaja — CHANGES_REQUESTED** [review]

**Comment by dawidjaja** [comment]
Do you think for those rooms that are vacant, is it better to write the date it's gonna be booked? So check for all the reservations. (This is kinda minor, if you have other things more important, just do that first)

File: src/main/java/seedu/address/model/Hotel.java (hunk 1 above)
Comment (sancharib-git) [reply src/main/java/seedu/address/model/Hotel.java:454]: I tried to do this and the list of rooms did not update in the UI when I checked in someone. (even if `model.updateFilteredRoomList()` is called. If you have any idea why this happens please share with me.

File: src/main/java/seedu/address/model/Hotel.java [hunk 3]
```diff
@@ -376,11 +376,13 @@ public int hashCode() {
...
         for (Booking booking : bookingList) {
-            if (stay.isInside(booking)) {
+            if (stay.getRoom().equals(booking.getRoom())
+                    && stay.getTimeTo().equals(booking.getTimeTo())) {
+                addStay(stay);
                 bookingList.remove(booking);
+                updateRoomStays();
             }
         }
-        addStay(stay);
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java:381]: I don't think you should remove this.

File: src/main/java/seedu/address/model/Hotel.java [hunk 4]
```diff
@@ -376,11 +376,13 @@ public int hashCode() {
      */
     public void checkIn(Stay stay) {
         for (Booking booking : bookingList) {
-            if (stay.isInside(booking)) {
+            if (stay.getRoom().equals(booking.getRoom())
+                    && stay.getTimeTo().equals(booking.getTimeTo())) {
+                addStay(stay);
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java]:
```suggestion
            if (stay.isInside(booking)) {
```

File: src/main/java/seedu/address/model/Hotel.java [hunk 5]
```diff
@@ -376,11 +376,13 @@ public int hashCode() {
...
                 bookingList.remove(booking);
+                updateRoomStays();
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java]:
```suggestion
```

File: src/main/java/seedu/address/model/Hotel.java [hunk 6]
```diff
@@ -376,11 +376,13 @@ public int hashCode() {
...
             }
         }
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java]:
```suggestion
        }
        addStay(stay);
        updateRoomStays();
```

File: src/main/java/seedu/address/model/Hotel.java [hunk 7]
```diff
@@ -438,4 +440,19 @@ public void addStay(Stay stay) {
...
+        for (Stay stay : stayList) {
+            for (Room room : roomList) {
+                if (stay.getRoom().equals(room)) {
+                    Room newRoom = new Room(room);
+                    newRoom.setStay(stay);
+                    roomList.setRoom(room, newRoom);
+                }
+            }
+        }
+    }
```
Comment (dawidjaja) [inline src/main/java/seedu/address/model/Hotel.java:459]:
```suggestion
    /**
     * Goes through stay list and sets
     * the current stay for each room
     */
    public void updateRoomStays() {
        for (Stay stay : stayList) {
            for (Room room : roomList) {
                if (stay.getRoom().equals(room)) {
                    Room newRoom = new Room(room);
                    newRoom.setStay(stay);
                    roomList.setRoom(room, newRoom);
                }
            }
        }
    }
```

**Review by dawidjaja — APPROVED** [review]
LGTM

**Pull request merged**